tests/utils-shared/src/main/java/org/keycloak/testsuite/util/oauth/AbstractOAuthClient.java

```diff
@@ -181,6 +182,18 @@ public TokenRevocationResponse doTokenRevoke(String token) {
         return tokenRevocationRequest(token).send();
     }
 
+    public TokenExchangeRequest tokenExchangeRequest(String subjectToken) {
+        return tokenExchangeRequest(subjectToken, OAuth2Constants.ACCESS_TOKEN_TYPE);
```

Comment:
Not sure if the default `subjectTokenType` to `ACCESS_TOKEN_TYPE` is what causes token-exchange V1 tests to fail? As token-exchange V1 is not compliant with the specification and it has `subject_token_type` as optional parameter. I guess there are some V1 tests, which are not using that parameter and that is the issue.

Reply:
That is already defaulting to access_token in main; https://github.com/keycloak/keycloak/blob/main/testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/util/oauth/TokenExchangeRequest.java#L22